src/accountsdb/download.zig

```diff
@@ -451,31 +463,22 @@ pub fn getOrDownloadAndUnpackSnapshot(
     }
 
     // check if we need to download a fresh snapshot
-    var accounts_db_exists = blk: {
-        if (validator_dir.openDir(sig.ACCOUNTS_DB_SUBDIR, .{ .iterate = true })) |dir| {
-            std.posix.close(dir.fd);
-            break :blk true;
-        } else |_| {
-            break :blk false;
+    if (std.fs.cwd().openDir(snapshot_path, .{ .iterate = true })) |dir| {
+        defer std.posix.close(dir.fd);
+        // clear old snapshots, if we will download a new one
+        if (force_new_snapshot_download) {
+            logger.info().log("deleting accounts_db dir...");
+            try dir.deleteTreeMinStackSize("accounts_db");
```

Comment:
Doesn't this change make it so this function is always forced to download a new snapshot, instead of being able to detect and return an existing one?

Reply:
only if`force_new_snapshot_download` is set, we delete the directory - otherwise we re-use it

Reply:
Yeah, that fixes it.